wallet/reference/non-evm-apis/starknet-snap-api.md

```diff
@@ -0,0 +1,867 @@
+---
+description: See the Starknet Snap API reference.
+sidebar_position: 1
+---
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+# Starknet Snap API
+
+When connected to the [Starknet Snap](../../how-to/use-non-evm-networks/starknet/index.md), dapps
+can use the Starknet Snap API to interact with users' Starknet accounts (for example, to send transactions).
+
+Currently, the [`get-starknet`](https://github.com/starknet-io/get-starknet) library only supports the
+following Starknet Snap API methods:
+
+- [`starkNet_extractPublicKey`](#starknet_extractpublickey)
+- [`starkNet_signMessage`](#starknet_signmessage)
+- [`starkNet_upgradeAccContract`](#starknet_upgradeacccontract)
+- [`starkNet_verifyMessage`](#starknet_verifymessage)
+
+The examples on this page use the
+[`wallet_invokeSnap`](/snaps/reference/wallet-api-for-snaps/#wallet_invokesnap) JSON-RPC method,
+which supports all Starknet Snap API methods.
+
+:::note
+
+You can also communicate with the Starknet network using the
+[Starknet API](/services/reference/starknet).
+
+:::
+
+## Supported networks
+
+Starknet currently supports two public networks.
+Use these networks' chain IDs with the Starknet Snap API methods.
+
+| Network           | Chain ID (Hexadecimal)   |
+|-------------------|--------------------------|
+| Mainnet           | `0x534e5f4d41494e`       |
+| Testnet (Sepolia) | `0x534e5f5345504f4c4941` |
+
+## API methods
+
+### `starkNet_createAccount`
+
+Deploys an account contract.
+
+#### Parameters
+
+- `addressIndex`: `integer` - (Optional) Specific address index of the derived key in
+  [BIP-44](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki).
+- `deploy`: `boolean` - (Optional) Indicate whether to include send the deploy transaction for the
+  account contract.
+  The default is `false`.
+- `chainId`: `string` - (Optional) ID of the target Starknet network.
+  The default is the Starknet Sepolia testnet.
+
+#### Returns
+
+The response from Starknet's `gateway/add_transaction` API endpoint.
```

Comment:
Since the `feeder_gateway`/`gateway` has been [deprecated](https://community.starknet.io/t/feeder-gateway-deprecation/100233), should this be updated?